libpromises/crypto.c

```diff
@@ -534,6 +534,14 @@ int DecryptString(char *out, size_t out_length, const char *in, int cipherlen,
     if (key == NULL)
         ProgrammingError("DecryptString: session key == NULL");
 
+    size_t max_plaintext_length = PlainTextLengthMax(CfengineCipher(type), cipherlen);
+
+    if(max_plaintext_length > out_length)
+    {
+        ProgrammingError("DecryptString: output buffer too small: max_plaintext_length (%ld) > out_length (%ld)",
+                          max_plaintext_length, out_length);
+    }
+
```

Comment:
Same here, this commit should be squashed together with:

* Add utiltiy function PlainTextLengthMax()...
* Rearrange parameters of DecryptString()...
* Pass output buffer size to DecryptString()...